tabrepo/benchmark/experiment/experiment_constructor.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import annotations
+
+import copy
+from typing import Type
+
+from autogluon.core.models import AbstractModel
+
+from tabrepo.benchmark.models.wrapper.abstract_class import AbstractExecModel
+from tabrepo.benchmark.models.wrapper.AutoGluon_class import AGSingleWrapper
+from tabrepo.benchmark.experiment.experiment_runner import ExperimentRunner, OOFExperimentRunner
+from tabrepo.utils.cache import AbstractCacheFunction, CacheFunctionDummy
+from autogluon_benchmark.tasks.task_wrapper import OpenMLTaskWrapper
+
+
+class Experiment:
+    """
+
```

Comment:
Could you add a docstring with the context you give in the example?

Reply:
added docstrings to Experiment + subclasses